packages/arb-util/core/logreader.go

```diff
@@ -90,16 +95,44 @@ func (lr *LogReader) getLogs(ctx context.Context) error {
 			time.Sleep(1 * time.Second)
 		}
 
-		if len(logs) > 0 {
-			err = lr.consumer.AddLogs(logs)
+		currentLogCount, err := lr.consumer.CurrentLogCount()
+		if err != nil {
+			return err
+		}
+
+		currentLogIndex := new(big.Int).Sub(currentLogCount, big.NewInt(1))
+
+		if len(deletedLogs) > 0 && firstDeletedIndex.Cmp(currentLogIndex) <= 0 {
+			// Existing logs to delete
+			deletedCount := new(big.Int).Sub(currentLogCount, firstDeletedIndex)
+			if deletedCount.Cmp(big.NewInt(int64(len(deletedLogs)))) != 0 {
+				logger.Warn().
+					Uint64("currentLogCount", currentLogCount.Uint64()).
+					Uint64("firstDeletedIndex", firstDeletedIndex.Uint64()).
+					Int("deletedLogs count", len(deletedLogs)).
+					Msg("more deleted logs sent than we previously received")
+			}
+			if err = lr.consumer.DeleteLogs(deletedLogs[:deletedCount.Uint64()]); err != nil {
+				return err
+			}
+
+			currentLogCount = firstDeletedIndex
+			err = lr.consumer.UpdateCurrentLogCount(currentLogCount)
 			if err != nil {
 				return err
 			}
 		}
 
-		if len(deletedLogs) > 0 {
-			err = lr.consumer.DeleteLogs(deletedLogs)
-			if err != nil {
+		if len(logs) > 0 {
+			cmp := firstIndex.Cmp(currentLogCount)
+			if cmp == 1 {
+				// Some logs were skipped, pull these manually
```

Comment:
How is it possible to end up in this case. Seems like maybe it should be an error

Reply:
I'm happy to treat it as an error